### PR TITLE
Tests for fs package.

### LINKED ISSUE #35: No test for client and fs package
Tests for client/client.go and fs/fs.go should be delivered. It seems that small code refactoring of these packages is needed to mock external lib methods.

### PR DESCRIPTION
**Pull request by PatrykMatyjasek**
Tests for fs package.
Fixes #35

Signed-off-by: Patryk Matyjasek <patryk.matyjasek@intel.com>

### CODE REVIEW

File: container/fs/fs_test.go [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+		So(dsm[DeviceId{Major: 8, Minor: 1}].IoInProgress, ShouldEqual, 0)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].IoTime, ShouldEqual, 330)
+		So(dsm[DeviceId{Major: 8, Minor: 1}].WeightedIoTime, ShouldEqual, 330)
+		So(err, ShouldBeNil)
+	})
+	Convey("Check getDiskStatsMap no file", s.T(), func() {
+		_, err := getDiskStatsMap("/tmp/proc/diskstat")
+		So(err, ShouldBeNil)
+	})
+	Convey("Check getDiskStatsMap no file", s.T(), func() {
```
Comment (katarzyna-z) [inline container/fs/fs_test.go]: Please change titles of two "Convey"s, now you have the same title for three "Convey"s.

File: container/fs/fs.go [hunk 2]
```diff
@@ -627,7 +634,17 @@ func parseDMStatus(dmStatus string) (uint64, uint64, error) {
...
 func getZfstats(poolName string) (uint64, uint64, uint64, error) {
-	dataset, err := zfs.GetDataset(poolName)
+	var dataset *zfs.Dataset
+	var err error
+	if poolName == "snapzfs" {
+		dataset = &zfs.Dataset{Name: poolName}
+		dataset.Used = 5000
+		dataset.Avail = 2000
+		dataset.Usedbydataset = 100
+
```
Comment (IzabellaRaulin) [inline container/fs/fs.go]: empty line

File: container/fs/fs_test.go [hunk 3]
```diff
@@ -0,0 +1,281 @@
+/*
```
Comment (IzabellaRaulin) [inline container/fs/fs_test.go]: please define test type

File: container/fs/fs_test.go [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+		So(usage, ShouldEqual, 0)
+		So(err, ShouldNotBeNil)
+	})
+	Convey("Check usage parser invalid data", s.T(), func() {
+		usage, err := diskUsage("cat", []string{"/tmp/test/usage_invalid_parse"})
+		So(usage, ShouldEqual, 0)
+		So(err, ShouldNotBeNil)
+	})
+	Convey("Check getDiskStatsMap", s.T(), func() {
+		dsm, err := getDiskStatsMap("/tmp/proc/diskstats")
```
Comment (IzabellaRaulin) [inline container/fs/fs_test.go]: So(err, ShouldBeNil) should be before the another checking

File: container/fs/fs_test.go [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+		})
+		Convey("Check root valid overlay driver", func() {
+			stats := container.NewStatistics()
+			err := du.GetStats(stats, container.GetStatOpt{"container_id": "root", "container_drv": "overlay", "procfs": "/tmp/proc", "root_dir": "/tmp"})
+			So(err, ShouldBeNil)
+		})
+		Convey("Check invalid root driver", func() {
+			stats := container.NewStatistics()
+			err := du.GetStats(stats, container.GetStatOpt{"container_id": "root", "container_drv": "ext", "procfs": "/tmp/proc", "root_dir": "/tmp"})
+			So(err, ShouldBeNil)
```
Comment (IzabellaRaulin) [inline container/fs/fs_test.go]: For `Check invalid root driver` I expected to get an error - am I right?

File: container/fs/fs_test.go [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+	usage := make(map[string]uint64)
+	du := DiskUsageCollector{DiskUsage: usage}
+	Convey("GetStats tests", s.T(), func() {
+		du.DiskUsage["/tmp"] = 2000
+		du.DiskUsage["/tmp/aufs/diff/27fa0900fe22"] = 250
+		du.DiskUsage["/tmp/overlay/27fa0900fe22"] = 929
+		Convey("Check invalid root dir", func() {
+			stats := container.NewStatistics()
+			err := du.GetStats(stats, container.GetStatOpt{"container_id": "27fa0900fe22", "container_drv": "aufs", "procfs": "/tmp/proc", "root_dir": "/invalid_dir"})
+			So(err, ShouldBeNil)
```
Comment (IzabellaRaulin) [inline container/fs/fs_test.go]: For `Check invalid root dir` I expected to get an error - am I right?

File: container/fs/fs_mocks.go [hunk 7]
```diff
@@ -0,0 +1,180 @@
+/*
```
Comment (IzabellaRaulin) [inline container/fs/fs_mocks.go]: Here also test type should be defined - to be clear that these mocks are used in small tests - see https://github.com/intelsdi-x/snap/blob/master/control/fixtures/fixtures.go#L1

**Pull request merged**